cypress/support/commands.js

```diff
@@ -16,24 +21,42 @@ Cypress.Commands.overwrite('visit', (orig, url, options) => {
   if (Cypress.env('SECURITY_ENABLED')) {
     let newOptions = options;
     if (options) {
-      newOptions['auth'] = {
-        username: 'admin',
-        password: 'admin',
-      };
+      newOptions['auth'] = ADMIN_AUTH;
     } else {
       newOptions = {
-        auth: {
-          username: 'admin',
-          password: 'admin',
-        },
+        auth: ADMIN_AUTH,
       };
     }
+    newOptions.qs = { security_tenant: 'private' };
     orig(url, newOptions);
   } else {
     orig(url, options);
   }
 });
 
+/**
+ * Overwite request command to support authentication similar to visit
+ */
+Cypress.Commands.overwrite('request', (originalFn, ...args) => {
+  let defaults = {};
+  if (Cypress.env('SECURITY_ENABLED')) {
+    defaults.auth = ADMIN_AUTH;
+  }
+
+  let options = {};
+  if (typeof args[0] === 'object' && args[0] !== null) {
+    options = Object.assign({}, args[0]);
+  } else if (args.length === 1) {
+    [options.url] = args;
+  } else if (args.length === 2) {
+    [options.method, options.url] = args;
+  } else if (args.length === 3) {
```

Comment:
can we add some comment for this args structure to support? method:url:body?

Reply:
added, thanks